usr.sbin/daemon/daemon.c

```diff
@@ -742,3 +743,20 @@ reopen_log(struct log_params *logparams)
 	logparams->output_fd = outfd;
 }
 
+static _Noreturn void
+daemon_terminate(struct daemon_state * state, int exitcode)
+{
+	if (state == NULL) {
```

Comment:
Can state be NULL?  If not, this should be an assert

Reply:
it should never be, this is simply defensive programming, can easily be removed in the current state of the code state can't be NULL

Reply:
An assert is the typical way to say 'this can't happen so tell me if it does' in FreeBSD.